### PR TITLE
chore(deprecation): removes all code deprecated before Elgg 1.9

### PR DESCRIPTION
**Pull request by juho-jaakkola**
With our current release policy (semver) Elgg 1.9 should had been 2.0, so I only removed the code deprecated in 1.8 or earlier.
- [x] list the removed functions/methods/views in upgrading.rst
- [x] Commit message should include `BREAKING CHANGE:` callout section

### CODE REVIEW

**Comment by mrclay** [comment]
I really think we should list all removed API resources. I guess particularly views because IDEs can't give any feedback about availability.

File: engine/lib/views.php [hunk 1]
```diff
@@ -801,14 +781,7 @@ function elgg_view_entity(\ElggEntity $entity, $vars = array(), $bypass = false,
...
 
-	if (is_array($vars)) {
-		$vars = array_merge($defaults, $vars);
-	} else {
-		elgg_deprecated_notice("Update your use of elgg_view_entity()", 1.8);
-		$vars = array(
-			'full_view' => $vars,
-		);
-	}
+	$vars = array_merge($defaults, $vars);
```
Comment (mrclay) [inline engine/lib/views.php]: Also add `array` type hints where appropriate.

File: engine/lib/views.php [hunk 2]
```diff
@@ -969,55 +942,24 @@ function elgg_view_annotation(\ElggAnnotation $annotation, array $vars = array()
  * @return string The rendered list of entities
  * @access private
  */
-function elgg_view_entity_list($entities, $vars = array(), $offset = 0, $limit = null, $full_view = true,
-$list_type_toggle = true, $pagination = true) {
-
-	if (!is_int($offset)) {
-		$offset = (int)get_input('offset', 0);
-	}
+function elgg_view_entity_list($entities, $vars = array()) {
```
Comment (mrclay) [inline engine/lib/views.php]: `array` type hint

**Comment by juho-jaakkola** [comment]
Array type hints have been added.

**Comment by juho-jaakkola** [comment]
All the removed stuff is now mentioned in upgrading.rst

File: docs/guides/upgrading.rst [hunk 3]
```diff
@@ -28,27 +28,35 @@ you will get an error message:
...
-All deprecated views and view arguments have been removed
----------------------------------------------------------
+List of deprecated views and view arguments that have been removed
+------------------------------------------------------------------
 
 We dropped support for and/or removed the following views:
 
+ * canvas/layouts/*
+ * categories
+ * categories/vuew
```
Comment (mrclay) [inline docs/guides/upgrading.rst]: typo

File: docs/guides/upgrading.rst [hunk 4]
```diff
@@ -68,7 +76,80 @@ We also dropped the following arguments to views:
  * "js" in icon views (e.g. icon/user/default)
  * "options" to input/radio and input/checkboxes which aren't key-value pairs
    will no longer be acceptable.
- 
+
+List of deprecated functions and methods that have been removed
+---------------------------------------------------------------
+
+Functions:
```
Comment (mrclay) [inline docs/guides/upgrading.rst]: Just use the already existing "Removed Functions" section?

Comment (mrclay) [inline docs/guides/upgrading.rst]: My thinking is, reading this, I wouldn't want multiple lists.

Comment (juho-jaakkola) [inline docs/guides/upgrading.rst]: Oops, I hadn't noticed the other list.

**Pull request merged**